### ISSUE TITLE
Add Integration Tests to the project.

### ISSUE THREAD

**Issue by naderghanbari** [vmunier/play-with-scalajs-example#40]
That would be great if this sample project had a few integration tests. That way it will be easier for people adopting Scalajs + Play + SBT combo. 

There are some caveats (I believe because the way `sbt-web` works: https://github.com/sbt/sbt-web/issues/54). But it would be nice to mention the best workaround here. 

So far I found one by using something like this in the server `sbt` descriptor:

      unmanagedResourceDirectories in IntegrationTest <+= baseDirectory(_ / "target/web/public/test"),
      resourceGenerators in IntegrationTest <+= PlayScalaJS.copyMappings(scalaJSDev, WebKeys.public in TestAssets).map(_ => Seq[File]())

**Comment by naderghanbari** [comment]
So far I found one by using something like this in the server `sbt` descriptor:

      unmanagedResourceDirectories in IntegrationTest <+= baseDirectory(_ / "target/web/public/test"),
      resourceGenerators in IntegrationTest <+= PlayScalaJS.copyMappings(scalaJSDev, WebKeys.public in TestAssets).map(_ => Seq[File]())

Please share your thoughts and let me know if there are better workarounds. If not, is a PR with this workaround welcome?

**Comment by vmunier** [comment]
There is an [integration test](https://github.com/vmunier/play-with-scalajs-example/blob/3c3678d8576241f700ad11ca98dbe992f4d06bfa/server/test/IntegrationSpec.scala) in this project.

I don't have any real experience running integration tests in Play.

How do you run your integration tests in Play? Are you using `it:test`? Do you add the tests in a different folder than `test`?

**Comment by naderghanbari** [comment]
Yes, what I have now is putting tests in a separate folder, namely `it` and then adding `IntegrationTest:DefaultSettings` to the build file. In that sense there is almost no difference from what you have here. But the problem is the `launcher` file and other js files (`fastOpt` or `jsDeps`) are not available (I believe if you test the actual functionality in integration tests they will fail). So I guess despite the fact that Integration Tests has no great support by `SBT`, adding a working configuration to this project makes it easier for developers to find a workaround.
